usbSerialForAndroid/src/main/java/com/hoho/android/usbserial/driver/GsmSerialDriver.java

```diff
@@ -0,0 +1,153 @@
+package com.hoho.android.usbserial.driver;
+
+import android.hardware.usb.UsbConstants;
+import android.hardware.usb.UsbDevice;
+import android.hardware.usb.UsbEndpoint;
+import android.hardware.usb.UsbInterface;
+import android.util.Log;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+
+public class GsmSerialDriver implements UsbSerialDriver{
+
+    private static final String TAG = "GsmSerialDriver";
+    private final UsbDevice mDevice;
+    private final List<UsbSerialPort> mPorts;
+
+    private UsbInterface mControlInterface;
+    private UsbInterface mDataInterface;
+
+    private UsbEndpoint mControlEndpoint;
+
+    private int mControlIndex;
+    private boolean mRts = false;
+    private boolean mDtr = false;
+    private static final int SET_CONTROL_LINE_STATE = 0x22;
+
+
+    @Override
+    public UsbDevice getDevice() {
+        return mDevice;
+    }
+
+    @Override
+    public List<UsbSerialPort> getPorts() {
+        return mPorts;
+    }
+    private static int countPorts(UsbDevice device) {
+        return device.getInterfaceCount();
+    }
+
+    public GsmSerialDriver(UsbDevice mDevice) {
+        this.mDevice = mDevice;
+        mPorts = new ArrayList<>();
+        int ports = countPorts(mDevice);

```

Comment:
openInt() always uses interface 0. Adding a single port should be sufficient here instead of logic cloned from CdCAdm driver